msal_extensions/persistence.py

```diff
@@ -119,9 +119,11 @@ def save(self, content):
             handle.write(self._dp_agent.protect(content))
 
     def load(self):
-        # type: () -> str
+        """Return decrypted content, or None if input is empty."""
+        # type: () -> Optional[str]
         with open(self._location, 'rb') as handle:
-            return self._dp_agent.unprotect(handle.read())
+            data = handle.read()
+        return self._dp_agent.unprotect(data) if data else None
```

Comment:
If data is None,  it means the cache is corrupt. Because even if you log out all the users, the cache file should not be empty. So you can handle this here, but I would also handle the more general case of "cache is corrupt" (i.e. MSAL fails to parse the cache) and delete the cache file / keychain item / keyring entry

Reply:
Agreed, it will be good to have a general recovery method in the case of cache being corrupt. Created an [issue](https://github.com/AzureAD/microsoft-authentication-extensions-for-python/issues/68) to track this.